### ISSUE TITLE
Coverband does not swallow JSON.parse errors

### ISSUE THREAD

**Issue by jumski** [danmayer/coverband#93]
I do not have a root cause why it cannot parse json (probably uploaded some invalid baseline to our staging server), but it does not swallow `JSON.parse` errors, throwing: 

```JSON::ParserError: 784: unexpected token at ':22,"20":22,"24":22}}'```

The error is not catched by coverband and makes webserver fail with 500 error.
It should probably be swallowed and reported in coverband log.

**Comment by danmayer** [comment]
hmm this might be related to upgrading from a previous version of coverband and the format changed... if you have more of a stacktrace, I can likely improve the handling and log how to resolve the issue. @jumski do you have a stacktrace or any other details.

**Comment by jumski** [comment]
this was a fresh install of coverband
i'll try to recover the stack trace so you can investigate,

but don't you think @danmayer that it should swallow all exceptions (maybe report them somewhere if possible) so it never causes 500 on production?

**Comment by danmayer** [comment]
Yeah that is a good point, we shouldn't cause any production errors... I can definitely swallow and log, which is what was happening previously so I must have missed something when moving to adapter stores. I can look into it but if you can get me a stack trace that would definitely help @jumski

**Comment by jumski** [comment]
I am unable to get to that stack trace at this point :( too much digging in teh aws console, sorry man

**Comment by lacco** [comment]
I was trying out coverband on our staging server, but after a couple of hours, requests and sidekiq jobs started to fail with `JSON::ParserError: unexpected hash close at line 1, column 215732 [parse.c:558]`. Not sure how this is happening, but somehow the JSON file becomes invalid. Might there be a problem when two processes try to write to the file at the same time? When I deleted the file, everything was fine for the moment but the problem appeared again after some time.

This is the stack trace: 

```
/gems/coverband-1.5.4/lib/coverband/adapters/file_store.rb:51 in parse
/gems/coverband-1.5.4/lib/coverband/adapters/file_store.rb:51 in existing_data
/gems/coverband-1.5.4/lib/coverband/adapters/file_store.rb:20 in save_report
/gems/coverband-1.5.4/lib/coverband/base.rb:109 in report_coverage
/gems/coverband-1.5.4/lib/coverband/middleware.rb:13 in call
/gems/newrelic_rpm-4.8.0.341/lib/new_relic/agent/instrumentation/middleware_tracing.rb:92 in call
[...]
```

I will give the Redis adapter a try.

**Comment by danmayer** [comment]
hmm interesting @lacco could you share more of your `coverband` config. I am working on a larger branch with major changes and I can try to make sure to cover this issue

**Comment by danmayer** [comment]
ok great to hear, the file-based tracker wasn't really intended for production, I will try to make things like this more obvious and less error prone in the next release.

**Comment by danmayer** [comment]
This was resolved in the 2.0 branch... and can't occur any more even due to config issues...
https://github.com/danmayer/coverband/pull/107

closing this out.

**Comment by danmayer** [comment]
@lacco / @jumski I would highly recommend taking a look at the 2.0 release if you were still looking at working with Coverband it has resolved a large number of issues.
